lib/src/main/java/co/nimblehq/alpine/lib/nfc/NfcReaderImpl.kt

```diff
@@ -0,0 +1,235 @@
+                    if (securityInfo is PACEInfo) {
+                        passportService.doPACE(
+                            bacKey,
+                            securityInfo.objectIdentifier,
+                            PACEInfo.toParameterSpec(securityInfo.parameterId)
+                        )
+                        paceSucceeded = true
+                    }
+                }
+            } catch (e: Exception) {
+                e.printStackTrace()
+            }
+            passportService.sendSelectApplet(paceSucceeded)
+            if (!paceSucceeded) {
+                try {
+                    passportService.getInputStream(PassportService.EF_COM).read()
+                } catch (e: Exception) {
+                    passportService.doBAC(bacKey)
+                }
+            }
+
+            val dg1InputStream = passportService.getInputStream(PassportService.EF_DG1)
+            val dg1File = DG1File(dg1InputStream)
+            val dg1MrzInfo = dg1File.mrzInfo
+
+            PassportInfo(
+                personDetails = getPersonDetails(dg1MrzInfo),
+                biometrics = getBiometrics(passportService),
+                additionalPersonDetails = getAdditionalPersonDetails(passportService),
+                additionalDocumentDetails = getAdditionalDocumentDetails(passportService),
+                documentType = DocumentType.from(dg1MrzInfo.documentCode),
+                documentPublicKey = getDocumentPublicKey(passportService)
+            )
+        } catch (e: Exception) {
+            e.printStackTrace()
+            null
+        }
+    }
+
+    private fun getPersonDetails(dg1MrzInfo: MRZInfo): PersonDetails {
+        return PersonDetails(
+            name = dg1MrzInfo.secondaryIdentifier.replace("<", " ").trim { it <= ' ' },
+            surname = dg1MrzInfo.primaryIdentifier.replace("<", " ").trim { it <= ' ' },
+            personalNumber = dg1MrzInfo.personalNumber,
+            gender = dg1MrzInfo.gender.toString(),
+            birthDate = dg1MrzInfo.dateOfBirth,
+            expiryDate = dg1MrzInfo.dateOfExpiry,
+            documentNumber = dg1MrzInfo.documentNumber,
+            nationality = dg1MrzInfo.nationality,
+            issuingState = dg1MrzInfo.issuingState
+        )
+    }
+
+    private fun getBiometrics(passportService: PassportService): Biometrics {
+        val faceImage = getFaceImage(passportService)
+        val portraitImage = getPortraitImage(passportService)
+        val signature = getSignature(passportService)
+        return Biometrics(
+            faceImageBitmap = faceImage?.bitmap,
+            faceImageBase64 = faceImage?.base64,
+            fingerprints = getFingerprints(passportService),
+            portraitImageBitmap = portraitImage?.bitmap,
+            portraitImageBase64 = portraitImage?.base64,
+            signatureBitmap = signature?.bitmap,
+            signatureBase64 = signature?.base64
+        )
```

Comment:
And this would end up as something like this 😄 
```suggestion
        return Biometrics(
            faceImage = getFaceImage(passportService),
            portraitImage = getPortraitImage(passportService),
            signatureImage = getSignature(passportService),
            fingerprints = getFingerprints(passportService)
        )
```